File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/Responder.java
package Auxiliary_Modules;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Random;

public class Responder {
    // simple random response dispenser
    private final ArrayList<String> responses = new ArrayList<>();
    private final Random rand = new Random();
    public Responder(String... replies) {
        Collections.addAll(responses, replies);
    }
    public String getAResponse(){
        if (responses.isEmpty()){
            return "";
        }
        return responses.get(rand.nextInt(responses.size()));
    }
    public Boolean responsesContainsStr(String str){
        return responses.contains(str);
    }
    public Boolean strContainsResponse(String str){
        boolean result = false;
        for (String tempStr: responses) {
            if (str.contains(tempStr)){
                result = true;
                break;
            }
        }
        return result;
    }
    public void addResponse(String s1){responses.add(s1);}
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/Responder.kt
package auxiliary_modules

import java.util.*

class Responder(vararg replies: String) {
    // simple random response dispenser
    private val responses = ArrayList<String>()
    private val rand = Random()

    init {
        Collections.addAll(responses, *replies)
    }

    val aResponse: String
        get() = if (responses.isEmpty()) {
            ""
        } else responses[rand.nextInt(responses.size)]

    fun responsesContainsStr(str: String): Boolean {
        return responses.contains(str)
    }

    fun strContainsResponse(str: String): Boolean {
        var result = false
        for (tempStr in responses) {
            if (str.contains(tempStr)) {
                result = true
                break
            }
        }
        return result
    }

    fun addResponse(s1: String) {
        responses.add(s1)
    }
}